### ISSUE TITLE
予約投稿

### ISSUE THREAD

**Issue by EbiseLutica** [misskey-dev/misskey#6318]
## Summary

- 日時を指定してその時間にノートをスケジュールする機能
- お知らせとかに使える

**Comment by kakkokari-gtyih** [comment]
予約投稿のノートを~~Redis~~DBの別テーブルに保管→QueueProcessorで投稿とかを思いついた（バックエンド詳しくないのでよくわかってない）

**Comment by syuilo** [comment]
Misskey本体の機能として実装すべきかは悩ましい

**Comment by kakkokari-gtyih** [comment]
わざわざそのために外部サービスを使うのもなんかアホらしい感じがする（Non-techなユーザーからすれば標準機能でほしいはず）

**Comment by syuilo** [comment]
- 手を加えるとこが多い割には需要が少なそう
- せっかくAPI公開してるから外部サービスに任せたい

**Comment by tassoman** [comment]
@EbiseLutica さん、misskeytoolsはこれを処理できますか。

**Comment by kakkokari-gtyih** [comment]
やっぱり予約投稿は内包したほうがいい気がしなくもない
ioのクリエイター増加もあるので一定の需要は見込める（クリエイター等だと、告知をぴったり0時に解禁したいなどのユースケースがある）

**Comment by tassoman** [comment]
I agree can be "temporarily" done with a 3rd party API client. But people must be aware of API tokens creation security concerns.  
I mean, if 3rd party App gets leaked, can be a problem for every misskey server. So, a massive token un-validator tool for administrators, should exists.

**Comment by KisaragiEffective** [comment]
https://misskey.io/notes/9uko0rpfufby0a3p

**Comment by Sayamame-beans** [comment]
revisit: ノートの下書き #15298 と似る部分があるため、そちらの実装待ちになっているという認識

**Comment by syuilo** [comment]
下書きが実装されたので実装のハードルは下がった

**Comment by syuilo** [comment]
どんな感じの設定UIにするか

**Comment by tassoman** [comment]
Soapbox, based on Pleroma, uses react datepicker calendar style. 
Be careful, when you use a datepicker, date should be localized, (MM/DD/YYYY Vs DD/MM/YYYY etc...) then finally sent to DB having TZ info.
I found this Vue DatePicker https://vue3datepicker.com/props/localization/
